pkg/apis/kops/validation/validation.go

```diff
@@ -1613,8 +1613,8 @@ func validateExternalDNS(cluster *kops.Cluster, spec *kops.ExternalDNSConfig, fl
 	}
 
 	if spec.Provider == kops.ExternalDNSProviderExternalDNS {
-		if cluster.IsGossip() {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("provider"), "external-dns does not support gossip clusters"))
+		if cluster.IsGossip() || cluster.UsesNoneDNS() {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("provider"), "external-dns does not support clusters without real DNS"))
```

Comment:
"without public DNS"?

Reply:
Doesn't it work also with private DNS, available only inside the VPC?

Reply:
I think "real dns" is a bit ambiguous. Perhaps separate messages for Gossip and NoneDNS?

Reply:
Sounds good to me.